zyte_api/utils.py

```diff
@@ -44,8 +44,8 @@ def _process_query(query):
     return {**query, "url": safe_url}
 
 
-def user_agent(library):
-    return 'python-zyte-api/{} {}/{}'.format(
-        __version__,
-        library.__name__,
-        library.__version__)
+def _user_agent(custom_user_agent):
+    user_agent = f'python-zyte-api/{__version__}'
+    if custom_user_agent:
+        return f'{user_agent}, {custom_user_agent}'
+    return user_agent
```

Comment:
I wonder if it wouldn’t make sense to just go for a constant:

```suggestion
USER_AGENT = f'python-zyte-api/{__version__}'
```

Dependent libraries can do:

```python
from zyte_api.utils import USER_AGENT as _USER_AGENT
from . import __version__

USER_AGENT = f"foo/{__version__} {_USER_AGENT}"
```

Reply:
@Gallaecio good suggestion, and in this case we will almost rely on the user-agent from other dependent libraries, right? And in `python-zyte-api` we can use this just like:
```
class AsyncClient:
    def __init__(self, *,
                 ...
                 user_agent: Optional[str] = None,
                 ):
        ...
        self.user_agent = user_agent
...
    retrying = retrying or self.retrying
        post = _post_func(session)
        auth = aiohttp.BasicAuth(self.api_key)
        headers = {
            'User-Agent': self.user_agent,
            'Accept-Encoding': 'br'
        }
```
Also in this case there is nothing to test in this module and we can remove them.

Reply:
> The only one concern here is that this dependent library can overwrite user_agent and we can lost `python-zyte-api` as the user agent.

I think that’s OK. If someone really wants to not mention the version of python-zyte-api, it’s up to them.